apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -16,7 +18,15 @@
 
 
 def tiledbsoma_build_index(
-    keys: np.typing.NDArray[np.int64],
+    keys: Union[  # type: ignore[type-arg]
+        np.typing.NDArray[np.int64],
+        pa.Array,
+        pa.IntegerArray,
+        pd.Series,
+        pd.arrays.IntegerArray,
+        pa.ChunkedArray,
+        list[int],
```

Comment:
not something we have to do in this change, but now that py3.7 is finally behind us, it might be worth going back and cleaning up all the `List[whatever]`s and turning them into `list[whatever]`